arc-mlir/arc-mlir-build

```diff
@@ -60,6 +72,8 @@ function llvm-cmake {
       return
     fi
 
+    dune-build
```

Comment:
As far as I can tell, this won't rebuild the OCaml parts when source files have changed. It should go in `llvm-build` or even better, figure out if there is a way to build Dune-based things from inside CMake, like we do with the cargo parts.

Reply:
I am quite lost when it comes to CMake 👀 I can move it into `llvm-build` but I'm not sure that I can figure out the CMake stuff